modules/http/src/main/java/org/mule/module/http/internal/listener/HttpResponseHeaderBuilder.java

```diff
@@ -62,6 +62,12 @@ else if (headerValue instanceof String[])
         }
         else
         {
+            if (headerValue == null)
+            {
+                LOGGER.warn("The header " + headerName + " is null. Removing it from the response.");
```

Comment:
Where is it removed? Is the header removed if you do this?
builder.addHeader("name", "not null");
builder.addHeader("name", null);

Reply:
Is not remove, the header is not added to the list.